src/laboratory-resource.ts

```diff
@@ -83,3 +84,7 @@ export function rejectLabOrder(orderId: string, comment: string, abortController
     },
   });
 }
+
+export const labDateRange = getGlobalStore<{ dateRange: Date[] }>('lab-date-range', {
```

Comment:
I think we should make this store name more specific since it's only purpose is for filtering based on a date range. So something like `lab-filter-date-range` or anything better could suffice. 
Am also curious why you chose to use the global store in this case. Using this will persist this date range even after leaving this specific page and navigating to pages unrelated to labs. Is this the intended behaviour @donaldkibet

Reply:
Hello @pirupius thanks for the review. If not for the global store, what other option is there that would scope that store to the specific page?

Reply:
@amosmachora thanks for the work on this. Let me get back to u on this later today as we wait for more feedback

Reply:
So my worry was about the persistence of the data in the global store, specifically at what point is this cleared. Does it clear when a user is logged out? 
If it's not cleared then that would mean, the next day the user will resume with the older date filter which i think shouldn't be the case. 
Based on a slack thread it seems the only guaranteed store that is cleared at the end of a session (or when a user is logged out) is the user store. There's an option of testing it's behavior to verify this and if at all this is the case then you might either need to add a way to clear it or opt for session storage.

Reply:
Had not thought about that to be honest. Thanks for pointing it out. I will work to see if a more robust solution is available then update this PR. I however feel like the global store should allow clearing out of the bo or atleat allow for passing a parameter that would allow easy opt in or opt out of the auto clearance feature.

Reply:
@amosmachora we can use `useContext` here to keep things simple. We might want to have the filter component and the table under the context provider. I think this approach might be better if we are not correctly using the global store as indicated by @pirupius